cpp/test/neighbors/ann_cagra_c.cu

```diff
@@ -126,3 +138,214 @@ TEST(CagraC, BuildSearch)
   cuvsCagraIndexDestroy(index);
   cuvsResourcesDestroy(res);
 }
+
+TEST(CagraC, BuildExtendSearch)
+{
+  // create cuvsResources_t
+  cuvsResources_t res;
+  cuvsResourcesCreate(&res);
+  cudaStream_t stream;
+  cuvsStreamGet(res, &stream);
+
+  raft::resources handle;
+
+  int32_t dimensions           = 16;
+  int32_t main_data_size       = 1024;
+  int32_t additional_data_size = 64;
+  int32_t num_queries          = 4;
+
+  // create random data for datasets and queries
+  rmm::device_uvector<float> random_data_d(
+    (main_data_size + additional_data_size + num_queries) * dimensions, stream);
+  rmm::device_uvector<int32_t> random_labels_d(
+    (main_data_size + additional_data_size + num_queries) * dimensions, stream);
+  raft::random::make_blobs(random_data_d.data(),
```

Comment:
Very nice to see this! And i"m glad this helped get past the bug, at least for now. Do we have a Github issue for the problem you encountered here? If not, do you mind creating one and referencing the issue number in a TODO comment in the code?

Reply:
Good idea - just created an issue and added it in a comment!